starknet_py/tests/e2e/client/gateway_test.py

```diff
@@ -65,6 +65,32 @@ async def test_get_declare_v2_transaction():
     )
 
 
+@pytest.mark.asyncio
+async def test_get_block_with_declare_v2():
+    client = GatewayClient(
+        net="https://external.integration.starknet.io"
+    )  # TODO: Replace this with fixture
+
+    block = await client.get_block(block_number=283364)
+
+    assert (
+        DeclareTransaction(
+            class_hash=0x4E70B19333AE94BD958625F7B61CE9EEC631653597E68645E13780061B2136C,
+            compiled_class_hash=0x711C0C3E56863E29D3158804AAC47F424241EDA64DB33E2CC2999D60EE5105,
+            sender_address=0x2FD67A7BCCA0D984408143255C41563B14E6C8A0846B5C9E092E7D56CF1A862,
+            hash=0x722B666CE83EC69C18190AAE6149F79E6AD4B9C051B171CC6C309C9E0C28129,
+            max_fee=0x38D7EA4C68000,
+            signature=[
+                0x6F3070288FB33359289F5995190C1074DE5FF00D181B1A7D6BE87346D9957FE,
+                0x4AB2D251D18A75F8E1AD03ABA2A77BD3D978ABF571DC262C592FB07920DC50D,
+            ],
+            nonce=1,
+            version=2,
+        )
+        in block.transactions
```

Comment:
I wanted to ask what about the transaction receipts in the block, but it seems we do not have a field for receipts in the block dataclass.

Reply:
Probably because of RPC specs

Reply:
Rpc only has `getBlockWithTxHashes` and `getBlockWithTxs` which includes only information similar to this from `get_transaction`. So no way to map these.